core/src/main/java/hudson/model/ListView.java

```diff
@@ -216,6 +227,23 @@ public boolean hasJobFilterExtensions() {
         return items;
     }
 
+    @Override
+    public SearchIndexBuilder makeSearchIndex() {
+        SearchIndexBuilder sib = new SearchIndexBuilder().addAllAnnotations(this);
+        sib.add(new CollectionSearchIndex<TopLevelItem>() {// for jobs in the view
+            protected TopLevelItem get(String key) { return getItem(key); }
+            protected Collection<TopLevelItem> all() { return getItems(); }
+            @Override
+            protected String getName(TopLevelItem o) {
+                // return the name instead of the display for suggestion searching
+                return o.getName();
+            }
+        });
+        // add the display name for each item in the search index
+        addDisplayNamesToSearchIndex(sib, getItems(false));
```

Comment:
Why is there `false` here?

Reply:
Actually that is a good suggestion. I would think you want to recursively search over getItems. I am changing it to true and testing to see if that is behavior we want